src/main/java/com/dev/cinema/controller/UserController.java

```diff
@@ -28,15 +30,19 @@ public UserController(UserService userService,
     }
 
     @GetMapping("/by-email")
-    public UserResponseDto findByEmail(@RequestParam String email) {
+    public UserResponseDto findByEmail(Authentication authentication) {
+        Object principal = authentication.getPrincipal();
+        String email = "";
+        if (principal instanceof UserDetails) {
+            UserDetails userDetails = (UserDetails) principal;
+            email = userDetails.getUsername();
+        }
         Optional<User> userByEmail = userService.findByEmail(email);
-        return userMapper.toDto(userByEmail.orElseThrow(()
-                -> new RuntimeException("User by email " + email
-                + " wasn't found")));
+        return userMapper.toDto(userByEmail.get());
     }
 
     @PostMapping
-    public void add(@RequestBody UserRequestDto userRequestDto) {
+    public void add(@RequestBody @Valid UserRequestDto userRequestDto) {
         userService.add(userMapper.toEntity(userRequestDto));
     }
```

Comment:
this method should not be here - we have `register` for this purpose

Reply:
done